bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -248,13 +248,18 @@ func (r *KubeadmConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		log.Info("Cluster infrastructure is not ready, waiting")
 		conditions.MarkFalse(config, bootstrapv1.DataSecretAvailableCondition, bootstrapv1.WaitingForClusterInfrastructureReason, clusterv1.ConditionSeverityInfo, "")
 		return ctrl.Result{}, nil
-	// Reconcile status for machines that already have a secret reference, but our status isn't up to date.
-	// This case solves the pivoting scenario (or a backup restore) which doesn't preserve the status subresource on objects.
-	case configOwner.DataSecretName() != nil && (!config.Status.Ready || config.Status.DataSecretName == nil):
-		config.Status.Ready = true
-		config.Status.DataSecretName = configOwner.DataSecretName()
-		conditions.MarkTrue(config, bootstrapv1.DataSecretAvailableCondition)
-		return ctrl.Result{}, nil
+		// For machines that already have a secret reference ensure config status is up-to-date and the bootstrapSecret is adopted.
```

Comment:
nit: I find using adopt kind of confusing (it means different things in different controllers), so what about being more specific?
```suggestion
		// For machines that already have a secret reference ensure config status is up-to-date and the bootstrapSecret has the expected ownersRefs.
```